app/views/galleries/_single.haml

```diff
@@ -2,23 +2,25 @@
 - attrs = {class: 'section-ordered', data: {id: character_gallery.id, order: character_gallery.section_order}} if local_assigns[:character_gallery]
 %tr.gallery-header{**attrs}
   %th.padding-10{class: (klass if defined? klass)}
-    = link_to gallery.name, gallery_path(gallery), class: 'gallery-title'
-    - if gallery.user_id == current_user.try(:id) && !defined? skip_forms
-      = link_to add_gallery_path(gallery), class: 'gallery-add' do
+    - link = gallery ? gallery_path(gallery) : user_gallery_path(id: 0, user_id: @user.id)
+    = link_to gallery ? gallery.name : 'Galleryless icons', link, class: 'gallery-title'
+    - if is_owner && !skip_forms
+      = link_to add_gallery_path(gallery || {id: 0}), class: 'gallery-add' do
         .link-box.action-new + Add Icons
-      = link_to edit_gallery_path(gallery), class: 'gallery-edit' do
-        .link-box.action-edit
-          = image_tag "icons/pencil.png"
-          Edit Gallery
-      = link_to gallery_path(gallery), method: :delete, data: { confirm: 'Are you sure you want to delete this gallery? (This will not delete the icons.)' }, class: 'gallery-delete' do
-        .link-box.action-delete x Delete Gallery
+      - if gallery
+        = link_to edit_gallery_path(gallery), class: 'gallery-edit' do
+          .link-box.action-edit
+            = image_tag "icons/pencil.png"
+            Edit Gallery
+        = link_to gallery_path(gallery), method: :delete, data: { confirm: 'Are you sure you want to delete this gallery? (This will not delete the icons.)' }, class: 'gallery-delete' do
+          .link-box.action-delete x Delete Gallery
     - unless local_assigns[:hide_minmax]
       .gallery-box.float-right.gallery-minmax{data: {id: gallery.id}} -
-    - if local_assigns[:character_gallery] && (gallery.user_id == current_user.try(:id) || current_user.try(:admin?))
+    - if local_assigns[:character_gallery] && (is_owner || current_user.try(:admin?))
       .float-right
         = image_tag "icons/arrow_up.png", class: "section-up disabled-arrow"
         = image_tag "icons/arrow_down.png", class: "section-down disabled-arrow"
-- groups = gallery.gallery_groups_data
+- groups = gallery.gallery_groups_data if gallery
```

Comment:
I'm surprised this and the next line don't cause an error on galleryless

Reply:
Did a quick test on my dev env in case the tests were missing something, but it seems to load galleryless fine
`present?` might do something Fancy?

Reply:
Shouldn't! Oh well.